### ISSUE TITLE
Type annotations limit usage to python >= 3.9

### ISSUE THREAD

**Issue by ljmc-github** [YiqunChen1999/attrmap#1]
I'm filling this issue following this stackoverflow question: [import attrmap gives TypeError: 'type' object is not subscriptable](https://stackoverflow.com/q/73333512/17676984).

It appears this package uses type hinting generics introduced in python 3.9 which stop it from being usable with earlier version of python that are still fairly commonly used.

https://github.com/YiqunChen1999/attrmap/blob/802f233cc80e741055bbc7db79140b6328bc0b40/attrmap/attrmap.py#L91-L96

If you are not using other recent features, you might consider either downgrading the type annotation to the [`typing` module](https://docs.python.org/3/library/typing.html) to allow down to python 3.6 (or even 3.5 I'm no sure anymore), or adding [`from __future__ import annotations`](https://docs.python.org/3/library/__future__.html) which forces the postponed evaluation of annotations as proposed in [PEP563](https://www.python.org/dev/peps/pep-0563) allowing down to python 3.7.

If you are using new features, please consider adding the requirement for the appropriate lowest python version to your package specification.

```py
import typing as T

# [...]

def keys(self) -> T.list[str]:
    keys = filter(
        lambda x: x.startswith(self._prefix), self.__dict__.keys()
    )
    keys = map(lambda x: x.replace(self._prefix, ""), keys)
    return list(keys)
```

```py
from __future__ import annotations

# [...]

def keys(self) -> list[str]:
    keys = filter(
        lambda x: x.startswith(self._prefix), self.__dict__.keys()
    )
    keys = map(lambda x: x.replace(self._prefix, ""), keys)
    return list(keys)
```

**Comment by YiqunChen1999** [comment]
Thank you so much for your suggestion. I will fix it and update the docs this weekend.

**Comment by YiqunChen1999** [comment]
The unintentionally introduced type hint bug is fixed, now `attrmap` is available for python>=3.6 in the latest version.

Some other bugs are fixed (`delattr` and `__init__`), please update to the latest version. 

The API reference is also available [here](https://attrmap.readthedocs.io/en/latest/attrmap.html).
